core/src/main/java/io/nixer/nixerplugin/core/detection/events/FailedLoginRatioEvent.java

```diff
@@ -8,14 +10,31 @@
     public static final String FAILED_LOGIN_RATIO_ACTIVATION = "FAILED_LOGIN_RATIO_ACTIVATION";
     public static final String FAILED_LOGIN_RATIO_DEACTIVATION = "FAILED_LOGIN_RATIO_DEACTIVATION";
 
-    private final String ratio;
-
     public FailedLoginRatioEvent(double ratio) {
```

Comment:
It is not clear what the nature of `ratio` is. We should add check for meaningful value passed here. 
Also some javadocs would be nice.

Reply:
I would not do validation here. Let producers and consumers deal with that as this is part of the business logic. Javadoc added.

Reply:
As discussed offline, validation is done on other layers.